.github/workflows/integration-test-slow.yml

```diff
@@ -155,7 +155,7 @@ jobs:
         if: inputs.test_environment != 'default'
         run: |
           pip install uv
-          uv pip install --system "pyyaml==5.3.1" docker-compose
+          uv pip install --system "pyyaml==5.3.1" "docker==6.1.3" docker-compose
```

Comment:
Feels like at the very least we should include a #TODO here to address this at some point? Or will this just work forever?

Reply:
The only way to fix this is to get rid of docker-compose (which is just a thin Python layer on top of the `docker-compose` shell commands anyway) and replace it with something else. We may also need to do that sooner rather than later if it turns out that's the reason why our Docker CI jobs are still failing.